bulk_site_creator/views.py

```diff
@@ -0,0 +1,327 @@
+        potential_course_sites_query = get_course_instance_query_set(
+            term_id, sis_account_id
+        ).filter(canvas_course_id__isnull=True,
+                 sync_to_canvas=0,
+                 bulk_processing=0).select_related('course')
+
+        # Check if a course group or department filter needs to be applied to queryset
+        # The value of 0 is for the default option of no selected Department/Course Group
+        if school_id == 'colgsas':
+            if course_group_id and course_group_id != '0':
+                course_group_name = CourseGroup.objects.get(course_group_id=course_group_id).name
+                potential_course_sites_query = potential_course_sites_query.filter(course__course_group=course_group_id)
+        else:
+            if department_id and department_id != '0':
+                department_name = Department.objects.get(department_id=department_id).name
+                potential_course_sites_query = potential_course_sites_query.filter(course__department=department_id)
+
+    else:
+        # Get all potential course instances for the selected term in the account
+        # Further filter by the selected course instances from the DataTable
+        potential_course_sites_query = get_course_instance_query_set(
+            term_id, sis_account_id
+        ).filter(canvas_course_id__isnull=True,
+                 sync_to_canvas=0,
+                 bulk_processing=0,
+                 course_instance_id__in=course_instance_ids).select_related('course')
+
+    if potential_course_sites_query.count() > 0:
+        job = JobRecord(
+            sis_account_id=sis_account_id,
+            user_id=user_id,
+            user_full_name=user_full_name,
+            user_email=user_email,
+            school=school_id,
+            term_id=term_id,
+            sis_term_id=sis_term_id,
+            term_name=term_name,
+            department_id=department_id,
+            department_name=department_name,
+            course_group_id=course_group_id,
+            course_group_name=course_group_name,
+            template_id=template_id,
+            template_name=template_name,
+            workflow_state="pending",
+        )
+
+        log_extra = {
+            'sis_account_id': sis_account_id,
+            'user_id': user_id,
+            'user_full_name': user_full_name,
+            'user_email': user_email,
+            'school': school_id,
+            'term_id': term_id,
+            'term_name': term_name,
+            'department_id': department_id,
+            'department_name': department_name,
+            'course_group_id': course_group_id,
+            'course_group_name': course_group_name,
+            'template_id': template_id
+        }
+        # Sanitized input for log statements.
+        term_id = str(term_id)
+        sis_account_id = str(sis_account_id)
+        logger.debug(f'Generating task objects for term ID {term_id} (term name {term_name}) '
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It looks like you are trying to use `QuerySet.extra` for constructing raw SQL queries using external inputs. This could make your code vulnerable to SQL injection attacks, which can lead to loss of critical data. We recommend you to use `params` or `select_params` for passing parameters to the query string.

[Learn more](https://cwe.mitre.org/data/definitions/89.html)

Similar issue at line numbers 309, 323, and 325.